### ISSUE TITLE
Failed to setup U++ on macOS 11.5.2

### ISSUE THREAD

**Issue by AllenDang** [ultimatepp/ultimatepp#42]
I spent whole day trying to figure out how to run at least one example, tried every possible search source, still failed.
I feel very sad now.

I'm using macOS 11.5.2. I can confirm that all required libraries are installed.

Here is my journey:

1. Download upp-macos-16056.tar.xz, unzip it, launch theide.
2. Granted every privileges it requires.
3. A popup shows asking me to download the source, click ok.
4. I saw the system console showed up with git clone ..., after 30 minutes, still there, I checked target folder, seems nothing is downloaded.
5. I closed system console. Open terminal and type git clone ... manually, it works well, so network and git works fine, it's not the issue.
6. I tried again, still stuck at the git clone step.
7. Cancel git clone, start theide directly. I can only see MyApps in Assemply list.
![截屏2021-08-31 下午3 59 25](https://user-images.githubusercontent.com/18550/131465405-c58494ff-c66e-45a5-a8a9-963c67f22f4d.png)

8. I tried to use umk to compile a example by "./umk ./examples AddressBook CLANG -v ./adbook". Error message is "Missing package(s): CtrlLib Report".

I thought it might be the nightly build's issue, so I went to sourceforge and downloaded the stable build.

Lanuch theide, it stuck at svn ..., wait 30 minutes, nothing happened. Cancel, and lanuch theide, samething happened, I can only see MyApps in Assemply list.
Ran "./umk ./examples AddressBook CLANG -v ./adboo", same error.

I'm so exhausted now... How can I make it work?

**Comment by klugier** [comment]
Hello, i downloaded upp-macos-16056.tar.xz and it seems that it works for me. First of all, the popup that shows that you need to download sources should never be present in valid installation.

Could you try to place all upp related files in following directory (It seems that ide internally is looking into that directory):
`~/upp`

So, in this directory you should have following files:
`~/upp/uppsrc`
`~/upp/examples`
`~/upp/theide` (TheIDE App)
...

At the beggining please use TheIDE as your default development tool  for U++. Then you could try to use umk.

I hope this should  help. On our side we will need to make sure that we are directory name agnostic. Thanks  for report!

**Comment by mirek-fidler** [comment]
First of all, sorry for suboptimal experience. Truth be told, MacOS version
is so far the least matured of all.

If you see the dialog in the screenshot, you are 80% there. I am not sure
why starting clone from the theide does not work (and even why it wants to
do that at all), but if you have theide working and sources cloned, all you
need is to configure assemblies manuall. That is done by right clicking the
left pane and add folders there:

[image: image.png]

**Comment by AllenDang** [comment]
@klugier Put upp folder to `~/upp` does fix everything. Thanks. I love the structure design of ultimatepp.
I'm seeking for a chance to port ultimate to rust, digging now...:P

**Comment by KungFuFurby** [comment]
> First of all, sorry for suboptimal experience. Truth be told, MacOS version is so far the least matured of all. If you see the dialog in the screenshot, you are 80% there. I am not sure why starting clone from the theide does not work (and even why it wants to do that at all), but if you have theide working and sources cloned, all you need is to configure assemblies manuall. That is done by right clicking the left pane and add folders there: [image: image.png]

Your image did not display properly here. I came here because of the same initial problem, and I'm on Catalina (10.15), and I downloaded the upp-macos-16115.tar.xz file for the binary distribution.

I actually have hidden folders visible on my end via toggling a system option: in order to give it a kick after the initial failed attempt (I ran into the same problem with git clone, so I manually did it, but I determined I had a copy of the intended code anyways: I'll keep the git copy on hand in case of updates), I just copied my installation to the home directory, and cleared out the u++ folder from `.config/u++`, then ran the program again from the `~/upp` directory and it successfully auto-initialized itself.

No need to reopen: just providing details on how to get it to reinitialize itself should something go wrong during the first go round.
